public/javascripts/SVLabel/src/SVLabel/onboarding/OnboardingStates.js

```diff
@@ -181,7 +181,56 @@ function OnboardingStates (compass, mapService, statusModel, tracker) {
             "transition": function () {
                 tracker.push('Onboarding_Transition', {onboardingTransition: "redo-rate-attribute-1"});
                 var severity = parseInt(this.getAttribute("value"), 10);
-                return severity === 2 ? "adjust-heading-angle-1" : "redo-rate-attribute-1"
+                return severity === 2 ? "tag-attribute-1" : "redo-rate-attribute-1"
+            }
+        }, "tag-attribute-1": {
+            "properties": {
+                "action": "AddTag",
+                "labelType": "CurbRamp",
+                "minHeading": headingRanges["stage-1"][0],
+                "maxHeading": headingRanges["stage-1"][1],
+                "maxLabelCount": 1
+            },
+            "message": {
+                "message": 'Every label includes optional tags that add descriptive information. Choose appropriate tags for every label you place! ' +
+                    '<span class="bold">Let\'s add the “points into traffic” tag,</span> since this ramp points into the street.<br>' +
+                    '<img src="' + svl.rootDirectory + "img/onboarding/RatingCurbRampQuality-severity-2-v2.gif" +
+                    '" class="width-75" style="margin: 5px auto;display:block;" alt="Adding the \'points into traffic\' tag">',
+                "position": "top-right",
+                "parameters": null
+            },
+            "panoId": panoId,
+            "annotations": null,
+            "transition": function () {
+                var completedRate = 5 / numStates;
+                statusModel.setMissionCompletionRate(completedRate);
+                statusModel.setProgressBar(completedRate);
+                tracker.push('Onboarding_Transition', {onboardingTransition: "tag-attribute-1"});
+                var tags = this.getProperty('tagIds');
+                return tags.includes(2) && tags.length === 1 ? "adjust-heading-angle-1" : "redo-tag-attribute-1" // Where 2 is the tag_id of the "points into traffic" tag
```

Comment:
@misaugstad Unfortunately, it would be pretty cumbersome. The `.getProperty('tagIds')` call returns an array of the tag_ids, and not of the tag values. So, we would have to somehow map the tag names to the tag ids. This is possible with the `labelTags` field of `ContextMenu.js`. However, the `labelTags` field is an array built from a JSON file and not some nice map where we could easily perform a lookup, so we'd have to iterate through the array every time we do a lookup.

You can see a similar problem occurring on lines 237-240 of `ContextMenus.js` (in `_handleTagClick()`), where we have to look up the tag name then use its ID. 

That being said, it isn't impossible. And, while inefficient and a little hard to read, it would be pretty easy to implement. Let me know if you think the trade-offs are worth it, and I can make the change.

Reply:
That is fine then, your commenting scheme is sufficient for now.